pkg/cmd/shellcmd/shell.go

```diff
@@ -0,0 +1,209 @@
+
+	newConfig := *config
+	newConfig.CurrentContext = ctxName
+
+	c.cleanup()
+	tmpDirName, err := ioutil.TempDir("/tmp", ".dx-shell-")
+	if err != nil {
+		return err
+	}
+	tmpConfigFileName := tmpDirName + "/config"
+	err = clientcmd.WriteToFile(newConfig, tmpConfigFileName)
+	if err != nil {
+		return err
+	}
+
+	shell := filepath.Base(os.Getenv("SHELL"))
+	prompt := c.createNewBashPrompt(os.Getenv("PS1"))
+	rcFile := defaultRcFile + "\nexport PS1=" + prompt + "\nexport KUBECONFIG=\"" + tmpConfigFileName + "\"\n"
+	tmpRCFileName := tmpDirName + "/.bashrc"
+
+	if shell == "zsh" {
+		prompt = c.createNewZshPrompt(os.Getenv("PS1"))
+		rcFile = zshRcFile + "\nexport PS1=" + prompt + "\nexport KUBECONFIG=\"" + tmpConfigFileName + "\"\n"
+		tmpRCFileName = tmpDirName + "/.zshrc"
+	}
+	err = os.WriteFile(tmpRCFileName, []byte(rcFile), 0600)
+	if err != nil {
+		return err
+	}
+
+	info := util.ColorInfo
+	fmt.Printf("Creating a new shell using the Kubernetes context %s\n", info(ctxName))
+	fmt.Printf("Shell RC file is %s\n\n", tmpRCFileName)
+	fmt.Printf("All changes to the Kubernetes context like changing environment, namespace or context will be local to this shell\n")
+	fmt.Printf("To return to the global context use the command: exit\n\n")
+
+	e := exec.Command(shell, "-rcfile", tmpRCFileName, "-i")
+	if shell == "zsh" {
+		env := os.Environ()
+		env = append(env, fmt.Sprintf("ZDOTDIR=%s", tmpDirName))
+		e = exec.Command(shell, "-i")
+		e.Env = env
+	}
+
+	return e.Run()
+}
+
+func (c *ShellCmd) filterContexts(config *api.Config) []string {
+	var contextNames []string
+	for k, v := range config.Contexts {
+		if k != "" && v != nil {
+			if c.Filter == "" || strings.Contains(k, c.Filter) {
+				contextNames = append(contextNames, k)
+			}
+		}
+	}
+
+	sort.Strings(contextNames)
+	return contextNames
+}
+
+func (c *ShellCmd) cleanup() {
+	//clean old folders
+	files, err := filepath.Glob("/tmp/.dx-shell-*")
+	if err != nil {
+		panic(err)
+	}
+	for _, f := range files {
+		if err := os.RemoveAll(f); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func (c *ShellCmd) PickContext(names []string, defaultValue string) (string, error) {
+	if len(names) == 0 {
+		return "", nil
+	}
+	if len(names) == 1 {
+		return names[0], nil
+	}
+	name := ""
+	prompt := &survey.Select{
+		Message: "Change Kubernetes context:",
+		Options: names,
+		Default: defaultValue,
+	}
+	err := survey.AskOne(prompt, &name, nil)
+	return name, err
+}
+
+func (c *ShellCmd) createNewBashPrompt(prompt string) string {
+	if prompt == "" {
+		return "'[\\u@\\h \\W \\$(jx prompt) ]\\$ '"
+	}
+	if strings.Contains(prompt, "jx prompt") {
+		return prompt
+	}
+	if prompt[0] == '"' {
+		return prompt[0:1] + "\\$(jx prompt) " + prompt[1:]
+	}
+	if prompt[0] == '\'' {
+		return prompt[0:1] + "$(jx prompt) " + prompt[1:]
+	}
+	return "'$(jx prompt) " + prompt + "'"
+}
+
+func (c *ShellCmd) createNewZshPrompt(prompt string) string {
+	if prompt == "" {
+		return "'[$(jx prompt) %n@%m %c]\\$ '"
```

Comment:
ummmmm

Reply:
yeah, I need to workout a solution to that

maybe will add a `dx prompt`?